src/wallet/receive.cpp

```diff
@@ -257,6 +257,10 @@ bool CachedTxIsFromMe(const CWallet& wallet, const CWalletTx& wtx, const isminef
 bool CachedTxIsTrusted(const CWallet& wallet, const CWalletTx& wtx, std::set<uint256>& trusted_parents)
 {
     AssertLockHeld(wallet.cs_wallet);
+
+    // This wtx is already trusted
+    if (trusted_parents.contains(wtx.GetHash())) return true;
```

Comment:
Is this optimization directly related to this PR, or did you take the opportunity to include it here?

Reply:
I believe it's necessary so that repeated calls to `CachedTxIsTrusted` that would now occur once for each TXO can be skipped early.